src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightsService.java

```diff
@@ -239,6 +269,30 @@ public void setExporter(final MetricType type, final Settings settings) {
         }
     }
 
+    /**
+     * Set search query metrics enabled to enable collection of search query categorization metrics
+     * @param searchQueryMetricsEnabled boolean flag
+     */
+    public void setSearchQueryMetricsEnabled(boolean searchQueryMetricsEnabled) {
+        this.searchQueryMetricsEnabled = searchQueryMetricsEnabled;
```

Comment:
In the `setSearchQueryMetricsEnabled` function, you might also need to check if we want to stop / restart the whole query insights service. We should write a generic function for this, like:
```
public void checkAndStopQueryInsights(...) {

}

    
public void checkAndRestartQueryInsights(...) {

}

 public void setSearchQueryMetricsEnabled(boolean searchQueryMetricsEnabled) {
    if (searchQueryMetricsEnabled) {
        ...
        checkAndRestartQueryInsights(...)
    } else {
        ...
        checkAndStopQueryInsights(...)
    }
 }
 public void setEnableTopQueries(final MetricType metricType, final boolean enabled) {
        ...
        if (!enabled) {
            ...
            checkAndStopQueryInsights(...)
        } else {
           ...
           checkAndRestartQueryInsights(...)
        }
    }
```

Reply:
Done in the latest commit